src/starknet_runner/runner.rs

```diff
@@ -0,0 +1,388 @@
+use std::collections::HashMap;
+
+use cairo_rs::{
+    hint_processor::hint_processor_definition::HintProcessor,
+    types::relocatable::{MaybeRelocatable, Relocatable},
+    vm::{
+        errors::vm_errors::VirtualMachineError,
+        runners::{
+            builtin_runner::BuiltinRunner,
+            cairo_runner::{CairoArg, CairoRunner, ExecutionResources},
+        },
+        vm_core::VirtualMachine,
+    },
+};
+use felt::Felt;
+use num_traits::ToPrimitive;
+
+use crate::{
+    business_logic::{
+        execution::execution_errors::ExecutionError,
+        fact_state::in_memory_state_reader::InMemoryStateReader, state::cached_state::CachedState,
+    },
+    core::syscalls::{
+        business_logic_syscall_handler::BusinessLogicSyscallHandler,
+        syscall_handler::{self, SyscallHandler, SyscallHintProcessor},
+    },
+};
+
+use super::starknet_runner_error::StarknetRunnerError;
+use cairo_rs::types::relocatable::MaybeRelocatable::Int;
+
+pub(crate) struct StarknetRunner {
+    pub(crate) cairo_runner: CairoRunner,
+    pub(crate) vm: VirtualMachine,
+    pub(crate) hint_processor:
+        SyscallHintProcessor<BusinessLogicSyscallHandler<CachedState<InMemoryStateReader>>>,
+}
+
+impl StarknetRunner {
+    pub fn new(
+        cairo_runner: CairoRunner,
+        vm: VirtualMachine,
+        hint_processor: SyscallHintProcessor<
+            BusinessLogicSyscallHandler<CachedState<InMemoryStateReader>>,
+        >,
+    ) -> Self {
+        StarknetRunner {
+            cairo_runner,
+            vm,
+            hint_processor,
+        }
+    }
+
+    pub fn run_from_entrypoint(
+        &mut self,
+        entrypoint: usize,
+        args: &[&CairoArg],
+    ) -> Result<(), ExecutionError> {
+        let verify_secure = true;
+        let args: Vec<&CairoArg> = args.iter().map(ToOwned::to_owned).collect();
+
+        Ok(self.cairo_runner.run_from_entrypoint(
+            entrypoint,
+            &args,
+            verify_secure,
+            &mut self.vm,
+            &mut self.hint_processor,
+        )?)
+    }
+
+    pub fn get_execution_resources(&self) -> Result<ExecutionResources, ExecutionError> {
+        Ok(self.cairo_runner.get_execution_resources(&self.vm)?)
+    }
+
+    pub fn get_return_values(&self) -> Result<Vec<Felt>, StarknetRunnerError> {
+        self.vm
+            .get_return_values(2)
```

Comment:
Why this value is set to 2?

Reply:
once the execution is done, there are two values that are fetched from memory and use to get the real return values that are going to be returned in the field `retdata` of `CallInfo`, the function `get_return_values` receives as parameters the amount of data that must fetch from memory and as we need to two aforementioned fields, we pass a 2 to it.